packages/discovery-provider/ddl/migrations/0107_add_email_tables.sql

```diff
@@ -0,0 +1,46 @@
+begin;
+
+-- Store encrypted emails with their metadata
+CREATE TABLE IF NOT EXISTS encrypted_emails (
+    email_id INTEGER PRIMARY KEY,
+    seller_user_id INTEGER NOT NULL,
+    encrypted_email TEXT NOT NULL  -- base64 encoded
+);
+
+COMMENT ON TABLE encrypted_emails IS 'Stores encrypted emails and their metadata for secure communication between sellers and buyers';
+COMMENT ON COLUMN encrypted_emails.email_id IS 'Unique identifier for each encrypted email record';
+COMMENT ON COLUMN encrypted_emails.seller_user_id IS 'User ID of the seller who owns the encrypted email';
+COMMENT ON COLUMN encrypted_emails.encrypted_email IS 'Base64 encoded encrypted email content';
+
+-- Store the encryption keys for sellers and grantees
+CREATE TABLE IF NOT EXISTS email_encryption_keys (
+    key_id INTEGER PRIMARY KEY,
+    seller_user_id INTEGER NOT NULL,
+    owner_key TEXT NOT NULL,  -- base64 encoded
+    UNIQUE(seller_user_id)  -- one owner key per seller
+);
+
+COMMENT ON TABLE email_encryption_keys IS 'Stores encryption keys for sellers to manage their encrypted emails';
+COMMENT ON COLUMN email_encryption_keys.key_id IS 'Unique identifier for each encryption key record';
+COMMENT ON COLUMN email_encryption_keys.seller_user_id IS 'User ID of the seller who owns the encryption key';
+COMMENT ON COLUMN email_encryption_keys.owner_key IS 'Base64 encoded encryption key owned by the seller';
+
+-- Store the encrypted keys for grantees
+CREATE TABLE IF NOT EXISTS email_grantee_keys (
+    grantee_key_id INTEGER PRIMARY KEY,
+    seller_user_id INTEGER NOT NULL,
+    grantee_user_id INTEGER NOT NULL,
+    encrypted_key TEXT NOT NULL,  -- base64 encoded
+    UNIQUE(seller_user_id, grantee_user_id)  -- one key per seller-grantee pair
```

Comment:
nit: could be compound pkey, `grantee_key_id` and `seller_user_id`

Reply:
think monotonic key  kinda addresses this? we still want uniqueness though so this can probably stay

Reply:
monotonic ids run the risk of data divergence between different nodes based on order of insertion (though that risk is somewhat small and inconsequential depending on usage). I think these are better pkeyed using grantee and seller anyway, and that will give us a default index (and uniqueness) too.

Reply:
could the other tables also be keyed on seller_user_id + owner_key/email_owner_user_id? monotonic id divergence seems scary

Reply:
ah didn't see discussion in other thread